src/webots/gui/WbSingleTaskApplication.cpp

```diff
@@ -130,9 +132,18 @@ void WbSingleTaskApplication::convertProto() const {
   // Generate a node structure
   WbNode::setInstantiateMode(true);
   WbNode *node = WbNode::regenerateProtoInstanceFromParameters(model, fields, true, "");
-  for (WbNode *subNode : node->subNodes(true))
-    if (dynamic_cast<WbBasicJoint *>(subNode))
+  for (WbNode *subNode : node->subNodes(true)) {
+    if (type == QString("URDF") && dynamic_cast<WbSolidReference *>(subNode))
+      cout << tr("Warning: Exporting a Joint node with a SolidReference endpoint (").toUtf8().constData()
+           << static_cast<WbSolidReference *>(subNode)->name().toStdString()
+           << tr(") to URDF is not supported.").toUtf8().constData() << endl;
```

Comment:
It is better not to split a `tr` string as they are supposed to be translated at some point:
(caution: not tested, not clang-formatted)
```suggestion
      cout << tr("Warning: Exporting a Joint node with a SolidReference endpoint (%1) to URDF is not supported.").arg(static_cast<WbSolidReference *>(subNode)->name()).toUtf8().constData() << endl;
```